### PR TITLE
feat(deps): Use jQuery 3.1.0 and sinon 1.17.5

### PR DESCRIPTION
**Pull request by shane-tomlinson**
jQuery 3.1.0 is ~10kb smaller than 1.11.1 and is Promises A+ compliant.
sinon 1.17.5 is required to use jQuery 3.1.0

@vladikoff - r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Not ready, closing!

File: bower.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
     "fxa-password-strength-checker": "https://github.com/mozilla/fxa-password-strength-checker.git#d73b3ade374607e2749ee375301b0a168008b16f",
     "html5shiv": "3.7.2",
     "JavaScript-MD5": "1.1.0",
-    "jquery": "1.11.1",
-    "jquery-modal": "https://github.com/kylefox/jquery-modal.git#5e9bdc3e4de703089fba1e38ddc0f62d7de9eff7",
+    "jquery": "3.1.0",
+    "jquery-modal": "https://github.com/shane-tomlinson/jquery-modal.git#0576775d1b4590314b114386019f4c7421c77503",
```
Comment (shane-tomlinson) [inline bower.json]: The version from my account allows jQuery 3.x. I have opened a patch [against the original](https://github.com/kylefox/jquery-modal/pull/177) with this change.

File: app/tests/spec/views/settings/delete_account.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ define(function (require, exports, module) {
         sinon.stub(view, 'getSignedInAccount', function () {
           return account;
         });
-        sinon.spy(notifier, 'trigger', function () { });
+        sinon.stub(notifier, 'trigger', function () { });
```
Comment (shane-tomlinson) [inline app/tests/spec/views/settings/delete_account.js]: The `spy` for `stub` never caused problems before. Curious.

File: app/tests/spec/views/base.js [hunk 3]
```diff
@@ -149,43 +149,6 @@ define(function (require, exports, module) {
             });
       });
 
-      it('shows success messages', function () {
```
Comment (shane-tomlinson) [inline app/tests/spec/views/base.js]: These tests are all moved to the afterVisible section.

File: app/scripts/lib/promise.js [hunk 4]
```diff
@@ -2,27 +2,11 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// monkey patch p to be able to convert jQuery XHR promises to our promises.
```
Comment (vladikoff) [inline app/scripts/lib/promise.js]: yay!

**Comment by shane-tomlinson** [comment]
Thanks for the review @vladikoff! Now to ditch the pieces of jQuery that we don't need by using a custom build!

**Pull request merged**